### PR TITLE
Modify edit command

### PR DESCRIPTION
**Pull request by lyndonlim27**
- Users had to retype all existing groups/tags in order to add in new groups/tags
- Edit command now has a new prefix o/ which represents an overwrite flag
- Excluding o/ prefix allows users to append their new groups/tags instead of overwriting it
- Including o/prefix allows users to overwrite existing groups/tags

### CODE REVIEW

**Comment by lyndonlim27** [comment]
> * Users had to retype all existing groups/tags in order to add in new groups/tags
> * Edit command now has a new prefix o/ which represents an overwrite flag
> * Excluding o/ prefix allows users to append their new groups/tags instead of overwriting it
> * Including o/prefix allows users to overwrite existing groups/tags

Prefix has been changed to m/ instead which represents merge
- By default, all existing values will be overwritten
- If prefix m/ is present, groups/tags will be added to a person instead of overwriting it

File: src/main/java/seedu/address/logic/commands/EditCommand.java [hunk 1]
```diff
@@ -82,7 +89,19 @@ public CommandResult execute(Model model) throws CommandException {
...
+            // If merge and tag prefix is present but input is empty
+            if (!editPersonDescriptor.getTags().isEmpty() && editPersonDescriptor.getTags().get().isEmpty()) {
+                throw new CommandException(MESSAGE_NO_TAG_ADDED);
+            }
+
+            // If merge and group prefix is present but input is empty
+            if (!editPersonDescriptor.getGroups().isEmpty() && editPersonDescriptor.getGroups().get().isEmpty()) {
+                throw new CommandException(MESSAGE_NO_GROUP_ADDED);
+            }
+        }
```
Comment (edures888) [inline src/main/java/seedu/address/logic/commands/EditCommand.java:103]: I'm thinking if the checking of the fields should be done in the EditCommandParser instead?

File: src/test/java/seedu/address/logic/commands/EditCommandTest.java [hunk 2]
```diff
@@ -140,26 +146,100 @@ public void execute_invalidPersonIndexFilteredList_failure() {
...
     public void execute_invalidGroupAdded_failure() {
         EditCommand editCommand = new EditCommand(INDEX_FIRST_PERSON,
-                new EditPersonDescriptorBuilder().withName(VALID_NAME_BOB).withGroups("INVALID").build());
+                new EditPersonDescriptorBuilder().withName(VALID_NAME_BOB).withGroups("INVALID").build(), false);
 
         assertCommandFailure(editCommand, model, Messages.MESSAGE_INVALID_GROUP);
     }
 
+    @Test
+    public void execute_mergeeNoTagInput_failure() {
```
Comment (edures888) [inline src/test/java/seedu/address/logic/commands/EditCommandTest.java:163]: Typo? Haha

**Review by edures888 — COMMENTED** [review]
The changes looks good. Great work with the tests too!

File: src/main/java/seedu/address/logic/commands/EditCommand.java (hunk 1 above)
Comment (edures888) [reply src/main/java/seedu/address/logic/commands/EditCommand.java:103]: Kind of nitpicking...

**Pull request merged**